java/base-client/src/main/java/no/bankaxept/epayment/client/base/accesstoken/ScheduledAccessTokenPublisher.java

```diff
@@ -108,12 +106,82 @@ public void shutDown() {
     @Override
     public void subscribe(Flow.Subscriber<? super String> subscriber) {
         var token = atomicToken.get();
-        if (token != null && token.getExpiry().isBefore(clock.instant()))
+        if (token != null && token.getExpiry().isAfter(clock.instant()))
```

Comment:
don't know how this happened

Reply:
Do we lack test coverage, perhaps?

Reply:
So the thing is.. We *DO* have a test.. But I think that the problem is that when we fetch a token, it's pushed to all subscribers without verifying the date, but when we subscribe to an already "cached" token, we do this check..
I changed the test to check twice, this will fail with the old code

Reply:
[/AccessTokenPublisherTest.java](https://github.com/techcloud0/bankaxept-epayment-development-kit/pull/57#diff-164fcdf7ebfb787858fc55714aa8996dc5b171c8947da3e8c4a79f240ad5a805)